edge-hub/core/src/Microsoft.Azure.Devices.Routing.Core/checkpointers/Checkpointer.cs

```diff
@@ -198,18 +202,28 @@ static string GetContextString(Checkpointer checkpointer)
 
         public static class Metrics
         {
+            static readonly ConcurrentDictionary<string, ulong> Counters = new ConcurrentDictionary<string, ulong>();
```

Comment:
I wonder if this is the best way to do it, or if we should add an Increment/Decrement methods to IMetricsGauge.. 
I see that IMeasureGaugeMetrics has an overload that allows you to Get the metric value before setting it - 
https://github.com/AppMetrics/AppMetrics/blob/features/4.2.0/src/Core/src/App.Metrics.Abstractions/Gauge/IMeasureGaugeMetrics.cs#L30
Maybe we can use that? 

Another option could be IMeasureCounterMetrics, but it doesn't seem to have a SetValue method - https://github.com/AppMetrics/AppMetrics/blob/features/4.2.0/src/Core/src/App.Metrics.Abstractions/Counter/IMeasureCounterMetrics.cs

Reply:
So, I think we want to use gauge. Counter, I think, is only for 'increasing' values like number of messages handled. Since we know that we need to set after cleanup and do decrements I am not sure Counter will work.

For the updating the interface, I think that is a great way of doing this, especially since the prometheus version of gauge already has these methods :)